src/modules/decklink/consumer_decklink.cpp

```diff
@@ -575,7 +575,7 @@ class DeckLinkConsumer
 					ssize_t size = stride * height;
 
 					// convert lower field first to top field first
-					if ( !progressive )
+					if ( !progressive && m_displayMode->GetFieldDominance() == bmdUpperFieldFirst )
```

Comment:
Today, it seems consumer_decklink assumes MLT field order is bottom field first, but it really all depends on the "top_field_first" property of mlt_frame, and the consumer property "top_field_first determines" the output field order (default 0). I think this is the only issue I have with this set of patches, but now it blocks merging the pull request.